d2/src/test/java/com/linkedin/d2/balancer/subsetting/DeterministicSubsettingStrategyTest.java

```diff
@@ -0,0 +1,208 @@
+        "test", 10, _state);
+
+    double[] weights = new double[100];
+    Arrays.fill(weights, 1D);
+
+    assertTrue(_deterministicSubsettingStrategy.isSubsetChanged(0));
+    Map<String, Double> pointsMap = constructPointsMap(weights);
+    Map<String, Double> weightedSubset1 = _deterministicSubsettingStrategy.getWeightedSubset(pointsMap, 0);
+
+    assertFalse(_deterministicSubsettingStrategy.isSubsetChanged(0));
+    pointsMap = constructPointsMap(weights);
+    Map<String, Double> weightedSubset2 = _deterministicSubsettingStrategy.getWeightedSubset(pointsMap, 0);
+
+    assertSame(weightedSubset1, weightedSubset2);
+
+    assertTrue(_deterministicSubsettingStrategy.isSubsetChanged(1));
+    pointsMap = constructPointsMap(weights);
+    Map<String, Double> weightedSubset3 = _deterministicSubsettingStrategy.getWeightedSubset(pointsMap, 1);
+
+    assertEquals(weightedSubset1, weightedSubset3);
+    assertNotSame(weightedSubset1, weightedSubset3);
+
+    assertFalse(_deterministicSubsettingStrategy.isSubsetChanged(1));
+    pointsMap = constructPointsMap(weights);
+    Map<String, Double> weightedSubset4 = _deterministicSubsettingStrategy.getWeightedSubset(pointsMap, 1);
+
+    assertSame(weightedSubset3, weightedSubset4);
+
+    Mockito.when(_deterministicSubsettingMetadataProvider.getSubsettingMetadata(_state))
+        .thenReturn(new DeterministicSubsettingMetadata(0, 19));
+    assertTrue(_deterministicSubsettingStrategy.isSubsetChanged(1));
+  }
+
+  @Test(dataProvider = "uniformWeightData")
+  public void testDistributionWithUniformWeight(int clientNum, int hostNum, int minSubsetSize)
+  {
+    double[] weights = new double[hostNum];
+    Arrays.fill(weights, 1D);
+    Map<String, Double> pointsMap = constructPointsMap(weights);
+
+    Map<String, Double> distributionMap = new HashMap<>();
+
+    for (int i = 0; i < clientNum; i++)
+    {
+      Mockito.when(_deterministicSubsettingMetadataProvider.getSubsettingMetadata(_state))
+          .thenReturn(new DeterministicSubsettingMetadata(i, clientNum));
+      _deterministicSubsettingStrategy = new DeterministicSubsettingStrategy<>(_deterministicSubsettingMetadataProvider,
+          "test", minSubsetSize, _state);
+      Map<String, Double> weightedSubset = _deterministicSubsettingStrategy.getWeightedSubset(pointsMap, 0);
+      assertTrue(weightedSubset.size() >= Math.min(minSubsetSize, hostNum));
+
+      for (Map.Entry<String, Double> entry: weightedSubset.entrySet())
+      {
+        distributionMap.put(entry.getKey(), distributionMap.getOrDefault(entry.getKey(), 0D) + entry.getValue());
+      }
+    }
+
+    double host0WeightSum = distributionMap.getOrDefault("host0", 0D);
+    for (double weightSum: distributionMap.values())
+    {
+      assertEquals(weightSum, host0WeightSum, DELTA_DIFF);
+    }
+  }
+
+  @Test(dataProvider = "differentWeightsData")
+  public void testDistributionWithDifferentWeights(int clientNum, double[] weights, int minSubsetSize)
```

Comment:
Should we also verify minSubsetSize requirement is met?

Reply:
Done. Added minSubsetSize verifier to the testDistributionWithDifferentWeights test